compliance_checker/cf/cf.py

```diff
@@ -2360,67 +2360,68 @@ def check_packed_data(self, ds):
         When data to be packed contains missing values the attributes that indicate missing values (_FillValue, valid_min,
         valid_max, valid_range) must be of the same data type as the packed data.
         """
-        
         ret_val = []
         reasoning = []
-        names = list(ds.dataset.variables.iterkeys())
         for name, var in ds.dataset.variables.iteritems():
-            if hasattr(var, 'add_offset') or hasattr(var, 'scale_factor'):
+            if hasattr(var, 'add_offset') and hasattr(var, 'scale_factor'):
                 valid = False
-                if hasattr(var, 'add_offset') and hasattr(var, 'scale_factor'):
-                    offset = getattr(var, 'add_offset', '')
-                    scale = getattr(var, 'scale_factor', '')
-
-                    data_size = np.s_[ds.dataset.variables[name].size]
-                    data_type_check = np.reshape(ds.dataset.variables[name],data_size)[0]
-                    #, type(ds.dataset.variables[name][0,0,0]) ,type(scale), type(offset)
-                    for x in xrange(data_size):
-                        if type(np.reshape(ds.dataset.variables[name],data_size)[x]) != 'numpy.ma.core.MaskedConstant':
-                            data_type_check = np.reshape(ds.dataset.variables[name],data_size)[x]
-                            break
-                    if type(data_type_check) == type(scale) == type(offset):
+                valid_missing_variable = False
```

Comment:
I think this needs to be set to `True` here, as I don't see anything in the code below that does so.  Either that, or this variable doesn't need to exist at all.